projects/laji/src/app/shared-modules/document-viewer/document-annotation/document-annotation.component.ts

```diff
@@ -435,8 +430,9 @@ export class DocumentAnnotationComponent implements AfterViewInit, OnChanges, On
 
   private move() {
     this.document = this.result[this.indexPagination].document;
-    this.uri = this.identifying ? this.result[this.indexPagination].documentId : this.result[this.indexPagination].document?.documentId;
-    this.highlight = this.identifying ? this.result[this.indexPagination].fullURL : this.result[this.indexPagination].unit?.unitId;
+    this.result.findIndex(i => (i.fullURL === this.highlight || i.unit?.unitId === this.highlight));
+    this.uri = (this.result[this.indexPagination].documentId || this.result[this.indexPagination].document?.documentId);
+    this.highlight = (this.result[this.indexPagination].fullURL || this.result[this.indexPagination].unit?.unitId);
```

Comment:
How about:
```
const documentOrImage = this.result[this.indexPagination].document || this.result[this.indexPagination]
```
then bellow use something like this:
```
this.uri = documentOrImage.documentId
this.highlight = documentOrImage.fullURL || documentOrImage.unit?.unitId

Reply:
@vixriihi i made a solution a little bit different than yours. But when you can check the code we can discuss if it's not good